pkg/asset/manifests/mco.go

```diff
@@ -0,0 +1,172 @@
+	if cfg := generateMCOCfg(installConfig.Config); cfg != nil {
+		cfgData, err := yaml.Marshal(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to create %s manifests from InstallConfig: %w", mco.Name(), err)
+		}
+
+		mco.FileList = []*asset.File{
+			{
+				Filename: mcoCfgFilename,
+				Data:     cfgData,
+			},
+		}
+	}
+
+	return nil
+}
+
+// Files returns the files generated by the asset.
+func (mco *MCO) Files() []*asset.File {
+	return mco.FileList
+}
+
+// Load returns false since this asset is not written to disk by the installer.
+func (mco *MCO) Load(asset.FileFetcher) (bool, error) {
+	return false, nil
+}
+
+func generateMCOCfg(installConfig *types.InstallConfig) *operatorv1.MachineConfiguration {
+	cfg := &operatorv1.MachineConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: operatorv1.SchemeGroupVersion.String(),
+			Kind:       "MachineConfiguration",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster",
+			Namespace: "openshift-machine-config-operator",
+		},
+	}
+
+	// Currently the only reason to write an install-time config is if machine managers are needed,
+	// otherwise don't write one and defer to default behavior.
+	if machineMgrs := machineManagers(installConfig); len(machineMgrs) > 0 {
+		cfg.Spec.ManagedBootImages = operatorv1.ManagedBootImages{MachineManagers: machineMgrs}
+		return cfg
+	}
+	return nil
+}
+
+func machineManagers(ic *types.InstallConfig) []operatorv1.MachineManager {
+	var cpImg, wImg bool
+	switch ic.Platform.Name() {
+	case aws.Name:
+		cpImg, wImg = awsBootImages(ic)
+	case gcp.Name:
+		cpImg, wImg = gcpBootImages(ic)
+	default:
+		// We do not need to consider other platforms, because default boot image management has not been enabled yet.
+		return nil
+	}
+
+	return bootImageMgmt(cpImg, wImg)
+}
+
+func awsBootImages(ic *types.InstallConfig) (cpImg bool, wImg bool) {
+	if dmp := ic.AWS.DefaultMachinePlatform; dmp != nil && dmp.AMIID != "" {
+		return true, true
+	}
+
+	if cp := ic.ControlPlane; cp != nil && cp.Platform.AWS != nil && cp.Platform.AWS.AMIID != "" {
+		cpImg = true
+	}
+
+	if w := ic.Compute; len(w) > 0 && w[0].Platform.AWS != nil && w[0].Platform.AWS.AMIID != "" {
+		wImg = true
+	}
+	return
+}
+
+func gcpBootImages(ic *types.InstallConfig) (cpImg bool, wImg bool) {
+	if dmp := ic.GCP.DefaultMachinePlatform; dmp != nil && dmp.OSImage != nil {
+		return true, true
+	}
+
+	if cp := ic.ControlPlane; cp != nil && cp.Platform.GCP != nil && cp.Platform.GCP.OSImage != nil {
+		cpImg = true
+	}
+
+	if w := ic.Compute; len(w) > 0 && w[0].Platform.GCP != nil && w[0].Platform.GCP.OSImage != nil {
+		wImg = true
+	}
+	return
+}
+
+func bootImageMgmt(customCPImg, customWImg bool) []operatorv1.MachineManager {
+	var machineManagers []operatorv1.MachineManager
+
+	if !(customCPImg || customWImg) {
+		// No custom images, so nothing to do here.
+		return nil
+	}
+
+	if customCPImg {
+		// TODO: when CPMS boot image management is implemented in MCO-1007.
```

Comment:
[lint][1]:

```console
pkg/asset/manifests/mco.go:157:2: SA9003: empty branch (staticcheck)
	if customCPImg {
	^
```

Personally, I'd drop this `if`, shift the comment to the commit message, and add a note to the MCO-1007 Jira ticket to ask the MCO folks to reach out to the installer if/when they get around to ControlPlaneMachineSet boot-image management.

[1]: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/9783/pull-ci-openshift-installer-main-golint/1933255862084702208#1:build-log.txt%3A36-38